### PR TITLE
add a new flight contrller CM4pilot

### PR DESCRIPTION
**Pull request by robinluojl**

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/ACNS-CM4pilot/hwdef.dat [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+
+# --------------------- save flash ----------------------
+include ../include/minimize_features.inc
+# only include ublox GPS driver
+#include ../include/minimal_GPS.inc
+
+# reduce max size of embedded params for apj_tool.py
+define AP_PARAM_MAX_EMBEDDED_PARAM 1024
+#define HAL_GENERATOR_ENABLED 0
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/ACNS-CM4pilot/hwdef.dat:205]: I'm guessing these have been swiped from somewhere else.  Are these really appropriate for the board you are producing?

Perhaps including "save_some_flash.inc" would be better.

**Review by peterbarker — CHANGES_REQUESTED** [review]
Is this a board you're going to release to market?

File: libraries/AP_HAL_ChibiOS/hwdef/ACNS-CM4pilot/hwdef.dat (hunk 1 above)
Comment (robinluojl) [reply libraries/AP_HAL_ChibiOS/hwdef/ACNS-CM4pilot/hwdef.dat:205]: I did learn to write this file from another hwdef file, and the compiled code is running fine on our hardware.
We will test "save_some_flash.inc" Thank you for your suggestion

File: Tools/AP_Bootloader/board_types.txt [hunk 2]
```diff
@@ -225,6 +225,8 @@ AP_HW_CUBEBLACK_PERIPH               1401
 AP_HW_PIXRACER_PERIPH                1402
 AP_HW_SWBOOMBOARD_PERIPH             1403
 
+AP_HW_CM4PILOT                       1828
```
[the remark this answers is not in the document]
Comment (robinluojl) [reply Tools/AP_Bootloader/board_types.txt:237]: yes, it is. it's a low-cost and small-size flight controller integrated with a CM4. we will give it a new ID.

[the remark this answers is not in the document]
Comment (robinluojl) [reply Tools/AP_Bootloader/board_types.txt:237]: @peterbarker  We created a new PR https://github.com/ArduPilot/ardupilot/pull/24072 and closed this one. It shouldn't have been closed. lost contact with you, it‘s my fault, I am sorry. We allocated two consecutive IDs, and update the relevant information. Would you mind merging them? We are worried that the IDs will be taken later. Thanks a lot!

**Review by robinluojl — COMMENTED** [review]
@peterbarker We created a new PR https://github.com/ArduPilot/ardupilot/pull/24072 and closed this one. It shouldn't have been closed. lost contact with you, it‘s my fault, I am sorry. We allocated two consecutive IDs, and update the relevant information. Would you mind merging them? We are worried that the IDs will be taken later. Thanks a lot!

**Pull request closed without merging**